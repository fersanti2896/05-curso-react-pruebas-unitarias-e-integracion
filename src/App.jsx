
export const App = () => {
  return ( <h2>Â¡Hola mundo!</h2> );
}

