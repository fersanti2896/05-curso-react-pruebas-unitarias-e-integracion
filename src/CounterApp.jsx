import { useState } from 'react';
import PropTypes, { func } from 'prop-types';

export const CounterApp = ({ value }) => {
    const [ counter, setCounter ] = useState( value );

    const handleAdd = () => {
        setCounter( counter + 1 );
    };

    const handlePop = () => {
        setCounter( counter - 1 );
    };

    const handleReset = () => {
        setCounter( value );
    };

    return (
        <>
            <h1>CounterApp</h1>
            <h2> Value: { counter } </h2>

            <button onClick={ handleAdd }> +1 </button>
            <button onClick={ handlePop }> -1 </button>
            <button onClick={ handleReset }> Reset </button>
        </> 
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 0
}


