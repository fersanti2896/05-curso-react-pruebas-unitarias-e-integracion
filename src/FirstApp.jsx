import PropTypes from 'prop-types';

const persona = {
    name: 'Fernando',
    lastname: 'Nicolás', 
    age: 26,
    profession: 'Programador'
}

const saludo = () => {
    return '¡Bienvenido al curso React!';
}

export const FirstApp = ({ title, subTitle, name } ) => {
    return ( 
        <>
            <h1 data-testid="test-title"> { title } </h1>
            <h2>{ subTitle }</h2>
            <h3>Mi nombre es: ¡{ name }!</h3>

            <code>{ JSON.stringify( persona ) }</code>
        </>
    );
}

/* Definiendo las Proptypes */
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
    name: PropTypes.string
}

/* Definiendo las DefaultProps */
FirstApp.defaultProps = {
    title: 'No hay ningún título',
    subTitle: 'No hay subtítulo',
    name: 'Fernando Nicolás'
} 