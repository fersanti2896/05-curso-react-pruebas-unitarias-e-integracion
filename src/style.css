
html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.1rem;
    padding: 70px;
}

button {
    font-size: 0.9rem;
    margin-right: 5px;
    padding: 7px;
}