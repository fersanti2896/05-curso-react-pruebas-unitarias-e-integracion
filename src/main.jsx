
import React from 'react';
import ReactDOM from 'react-dom/client';
import { App } from './App';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';

import './style.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
      <FirstApp title={ 'Â¡Bienvenido al curso React!' } subTitle={ 'Primeros pasos en React' } />
      {/* <CounterApp value={ 10 }></CounterApp> */}
    </React.StrictMode>,
);