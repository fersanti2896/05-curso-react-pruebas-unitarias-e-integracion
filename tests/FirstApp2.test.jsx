import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en componente <FirstApp />', () => {
    const title = '¡Bienvenido al curso de React!';
    const subTitle = 'Primeros pasos';

    test('Debe hacer match con el snapshot.', () => {
        const { container } = render(<FirstApp title={ title }/>);

        expect( container ).toMatchSnapshot();
    });

    test('Debe mostrar el msg "¡Bienvenido al curso de React!".', () => {
        render(<FirstApp title={title}/>);

        expect( screen.getByText(title) ).toBeTruthy();
    });

    test('Debe mostrar el título en un h1', () => {
        render(<FirstApp title={title}/>);

        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title );
    });

    test('Debe de mostrar el subtítulo enviado en las props', () => {
        render(<FirstApp subTitle={subTitle}/>);

        expect( screen.getAllByText(subTitle).length ).toBe(1);
    })
});