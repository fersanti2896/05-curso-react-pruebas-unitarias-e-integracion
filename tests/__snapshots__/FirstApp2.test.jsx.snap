// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <FirstApp /> Debe hacer match con el snapshot. 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    ¡Bienvenido al curso de React!
     
  </h1>
  <h2>
    No hay subtítulo
  </h2>
  <h3>
    Mi nombre es: ¡
    Fernando Nicolás
    !
  </h3>
  <code>
    {"name":"Fernando","lastname":"Nicolás","age":26,"profession":"Programador"}
  </code>
</div>
`;
